File: knowledge-gap-platform/src/main/java/com/knowledgegap/knowledge_gap_platform/service/UserService.java
package com.knowledgegap.knowledge_gap_platform.service;

import com.knowledgegap.knowledge_gap_platform.dto.UpdateProfileRequest;
import com.knowledgegap.knowledge_gap_platform.dto.UserResponse;

import java.util.List;

public interface UserService{

    List<UserResponse> getAllUsers();
     UserResponse getUser(Long id);
     UserResponse updateUser(Long id, UpdateProfileRequest request);
     void delete(Long id);
     List<UserResponse> findLast7ByCreatedAtDesc();
}

File: knowledge-gap-frontend/src/services/userService.js
import api from "./api";

export const getUser = async (id) => {
    const response = await api.get(`/api/users/${id}`);
    return response.data;
};

export const updateUser = async (id, data) => {
    const response = await api.put(`/api/users/${id}`, data);
    return response.data;
};

export const getAllUsers = async () => {
    const response = await api.get("/api/users");
    return response.data;
};